web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\usergreeting;
+
+use Drupal\Component\Datetime\TimeInterface;
+
+/**
+ * Provides a greeting service for users dependent on the time of  day.
+ */
+class GreetingTime {
+
+  /**
+   * The user account.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $timeOutput;
+
+  /**
+   * GreetingTime constructor.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $timeOutput
+   *   The timestamp output.
+   */
+  public function __construct(TimeInterface $timeOutput) {
+    $this->timeOutput = $timeOutput;
+  }
+
+  /**
+   * Returns the greeting message.
+   *
+   * @return \Drupal\usergreeting\greetingMessage
+   *   The greeting message output.
+   */
+  final public function greetingMessage(string $greeting = NULL): greetingMessage {
+    $time_output = (int) date_default_timezone_get('H', $this->timeOutput->getRequestTime());
```

Comment:
Unfortunately this will still not return the timezone of the current user. However I also noticed that the whole implementation could be better structured.

Instead of injecting the AccountInterface service to the Block plugin, why not do it here? Then, in the block you'd only need to get the message and render it, no need for additional actions.

Also, creating a string with whitespace at the end is not a good practice (line 41, 44 and 47). There are plenty of ways to join strings with a delimiter. (hint: https://www.php.net/manual/en/function.implode.php)